packages/query-engine/source/controllers/query-engine-controller.ts

```diff
@@ -1,47 +1,86 @@
+const executeMySQL = async (
     queryRequest: ExecuteParameters,
-    next: NextFunction,
+    query: Query,
+    resource: Resource,
 ): Promise<any> => {
-    const query: Query = await QueryTemplateModel.findOne({
-        name: queryRequest.name,
-    }).exec();
+    const mySQLConfig: MySQLConfiguration = resource.mysql;
+    const config: Knex.Config = {
+        client: "mysql",
+        connection: {
+            host: mySQLConfig.host,
+            port: mySQLConfig.port,
+            user: mySQLConfig.databaseUserName,
+            password: mySQLConfig.databasePassword,
+            database: mySQLConfig.databaseName,
+        },
+    };
+    const knexInstance = knex(config);
 
-    const resource: Resource = await ResourceModel.findOne({
-        _id: query.resource,
-    }).exec();
+    if (typeof query.content === "object") {
+        const queryContent: any = query.content;
 
-    const mySQLConfig: MySQLConfiguration = resource.mysql;
+        let queryResult;
+        if (queryContent.hasOwnProperty("select") && queryContent.select) {
+            queryResult = await knexInstance
+                .select(queryContent.select)
+                .from(queryContent.from)
+                .where(queryContent.where);
+        }
 
-    const connection = await mysql.createConnection({
-        host: mySQLConfig.host,
-        port: mySQLConfig.port,
-        user: mySQLConfig.databaseUserName,
-        password: mySQLConfig.databasePassword,
-        database: mySQLConfig.databaseName,
-        namedPlaceholders:
-            typeof queryRequest.variables === "object" ? true : false,
-    });
-
-    let queryResult: any;
-    try {
-        const [results, fields] = await connection.execute(
+        if (queryContent.hasOwnProperty("insert") && queryContent.insert) {
+            queryResult = await knexInstance(queryContent.table)
+                .insert(queryContent.insert)
+                .where(queryContent.where);
+        }
+
+        if (queryContent.hasOwnProperty("update") && queryContent.update) {
+            queryResult = await knexInstance(queryContent.table)
+                .update(queryContent.update)
+                .where(queryContent.where);
+        }
+
+        return queryResult[0];
+    } else {
+        const queryResult = await knexInstance.raw(
             query.content,
             queryRequest.variables,
         );
-        queryResult = { results, fields };
-    } catch (error) {
-        next(error);
+        return queryResult[0];
     }
+};
 
-    return queryResult;
+const execute = async (queryRequest: ExecuteParameters): Promise<any> => {
+    try {
+        const query: Query = await QueryTemplateModel.findOne({
+            name: queryRequest.name,
+        }).exec();
+
+        const resource: Resource = await ResourceModel.findOne({
+            _id: query.resource,
+        }).exec();
+
+        console.log(query);
+
+        switch (resource.type) {
+            case "mysql": {
```

Comment:
We can include other SQL resource types that are inherently supported by Knex here.